hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -406,6 +407,9 @@ public void onResponse(Data response) {
                 public void onFailure(Throwable t) {
                 }
             });
+            if (getMapConfig().isStatisticsEnabled()) {
```

Comment:
yeah, I know that I am creating an extra object for every update but I don't have a better solution for this bug. Just like the qusp team, core team goes to great lengths to fix bugs and solve issues.